brain-marks/Categories/CategoryListViewModel.swift

```diff
@@ -36,4 +36,23 @@ final class CategoryListViewModel: ObservableObject {
     func editCategoryName(category: AWSCategory, newName: String) {
         DataStoreManger.shared.editCategory(category: category, newName: newName)
     }
+    
+    func pinCategory(category: AWSCategory){
+        var categoryIndex = -1;
+        for (index, each) in categories.enumerated() {
+            if(each.id == category.id){
+                categoryIndex = index
+            }
+        }
+        if(categoryIndex != -1){
+            categories.remove(at: categoryIndex)
+            categories.insert(category, at: 0)
+            for category in categories {
```

Comment:
@kiennguyenchi While this does work to pin categories to the top, it can't be implemented like this.

Calling `DataStoreManager.shared.deleteCategory()` also deletes the tweets inside that category because
 of how the data is laid out.